tests/h/settings_test.py

```diff
@@ -3,98 +3,104 @@
 from __future__ import unicode_literals
 
 import pytest
-import mock
+import logging
 
 from h import settings
 
 
-class FakeSetting(object):
-    def __init__(self, result):
-        self.result = result
+def asutf8(string):
+    return string.encode()
 
-    def __call__(self, environ):
-        return self.result
 
-    def __str__(self):
-        return 'fake setting'
-
-
-class TestDeprecatedSetting(object):
-    def test_emits_warnings_when_child_setting_is_used(self):
-        func = settings.DeprecatedSetting(FakeSetting(result={'foo': 'bar'}),
-                                          message='what to do instead')
-        func.warn = mock.Mock(spec_set=[])
-
-        result = func({})
-
-        assert result == {'foo': 'bar'}
-        func.warn.assert_called_once_with('use of fake setting is '
-                                          'deprecated: what to do instead')
+def test_database_url():
+    url = 'postgres://postgres:1234/database'
+    expected = 'postgresql+psycopg2://postgres:1234/database'
 
-    def test_emits_no_warnings_when_unused(self):
-        func = settings.DeprecatedSetting(FakeSetting(result=None),
-                                          message='what to do instead')
-        func.warn = mock.Mock(spec_set=[])
+    assert settings.database_url(url) == expected
 
-        result = func({})
 
-        assert result is None
-        assert not func.warn.called
+class TestSettingsManager(object):
+    def test_does_not_warn_when_deprecated_setting_is_not_used(self, caplog):
+        with caplog.at_level(logging.WARN):
+            settings_manager = settings.SettingsManager({}, {})
+            settings_manager.set('foo', 'FOO', deprecated_msg='what to do instead')
+        assert not caplog.records
 
+    def test_sets_value_when_deprecated_setting_is_used(self):
+        settings_manager = settings.SettingsManager({}, {'FOO': 'bar'})
+        settings_manager.set('foo', 'FOO', deprecated_msg='what to do instead')
 
-def asutf8(str):
-    return str.encode()
+        result = settings_manager.settings['foo']
```

Comment:
Should the tests be accessing `settings_manager.settings` like this? I thought the contract of `SettingsManager` was that `config.py` passes in `request.registry.settings` and `SettingsManager` modifies it? So the test should hold its own reference to the settings dict that it passed in, and should test that the dict was modified.